registry/spec.md

```diff
@@ -1491,71 +1632,1350 @@ An attempt to delete all versions MUST generate an error.
+scenarios:
 
-The Registry model defined by an Endpoint Registry is:
-```
+``` JSON
 {
-  "groups": [
-    {
-      "singular": "endpoint",
-      "plural": "endpoints",
-      "schema": "TBD",
-      "resources": [
-        {
-          "singular": "definition",
-          "plural": "definitions",
-          "versions": 0,
-          "mutable": true
+  "$schema": "https://cloudevents.io/schemas/registry",
+  "specversion": "0.4-wip",
+  "id": "urn:uuid:3978344f-8596-4c3a-a978-8fc9a6a469f7",
+  "endpoints" : 
+  {
+    "com.example.telemetry" : {
+      "type": "endpoint",
+      "id" : "com.example.telemetry",
+      "usage": "consumer",
+      "config": {
+        "protocol": "MQTT/5.0",
+        "strict": false,
+        "endpoints": [
+          "mqtt://mqtt.example.com:1883",
+        ],
+        "options" : {
+          "topic": "{deviceid}/telemetry"
         }
+      },
+      "format" : "CloudEvents/1.0"
+      "definitiongroups" :[
+        "#/definitiongroups/com.example.telemetryEvents"
       ]
-    },
-    {
-      "singular": "defintionGroup",
-      "plural": "definitionGroups",
-      "schema": "TBD",
-      "resources": [
-        {
-          "singular": "definition",
-          "plural": "definitions",
-          "versions": 0,
-          "mutable": true
+    }
+  },
+  "definitiongroups" : {
+    "com.example.telemetryEvents" : {
+      "type" : "definitiongroup",
+      "id" : "com.example.telemetryEvents",
+      "definitions": {
+        "com.example.telemetry" : {
+          "type": "cloudevent",
+          "id": "com.example.telemetry",
+          "description": "device telemetry event",
+          "format": "CloudEvents/1.0",
+          "metadata": {
+            "attributes": {
+              "id": {
+                "type": "string",
+                "required": true
+              },
+              "type": {
+                "type": "string",
+                "value": "com.example.telemetry",
+                "required": true
+              },
+              "time": {
+                "type": "datetime",
+                "required": true
+              },
+              "source": {
+                "type": "uritemplate",
+                "value": "{deploymentid}/{deviceid}",
+                "required": true
+              }
+            }
+          },
+          "schemaformat": "Protobuf/3.0",
+          "schemaurl" : "#/schemagroups/com.example.telemetry/schema/com.example.telemetrydata/versions/1.0"
         }
-      ]
+      }
+  },
+  "schemagroups" : {
+    "com.example.telemetry" : {
+      "type" : "schemagroup",
+      "id" : "com.example.telemetry",
+      "schemas": {
+        "com.example.telemetrydata" : {
+          "type": "schema",
+          "id": "com.example.telemetrydata",
+          "description": "device telemetry event data",
+          "format": "Protobuf/3.0",
+          "versions" : {
+            "1.0" : {
+              "type" : "schemaversion",
+              "id" : "1.0",
```

Comment:
should this be "version" instead? There are other required fields so we need to decide if all of them should be required or optional.

Reply:
No. It's the "id" of the schemaVersion object.

Reply:
I probably wasn't clear. It seems like the key on this line should be "version" or the value should be "com.example.telemetrydata". I would have expected the fields of the latest version to match the fields of the resource (sibling fields to "versions") - including the "id". Meaning, the child of "1.0" is the same resource as the child of "com.example.telemetrydata"  - but w/o the "versions" field.